### DISCUSSION TITLE
@xstate/svelte and useActor

### DISCUSSION THREAD

**Discussion by nikkez** [statelyai/xstate#1988] [Q&A]
Hi!

I've been using xstate for a while now with Typescript and Svelte. I really enjoy using ``useMachine`` from ``@xstate/svelte``. Now I have a case where I'm spawning machines in actions. I just can't figure out how to pass the actor to a Svelte component so it could use its machines state and interpreter so I could check the state and send events.

I noticed that there is ``useActor`` in ``@xstate/react`` which accepts the ``actor`` as parameter and returns ``state`` and ``send``.

I'm not proficient enough in Typescript so that I could try to copy the code from ``useActor`` in ``@xstate/react`` to ``@xstate/svelte`` and try it out.

Has somebody already been working on this or is the ``useActor`` feature planned to be implemented also in ``@xstate/svelte``?

**Comment by davidkpiano** [comment]
I would gladly accept a PR for this!

**Comment by santicros** [comment]
@xstate/vue is also missing the useActor and latest hooks for React, so would be interesting to port that too! 👌

**Reply by Andarist** [reply]
@santicros We would gladly accept a PR for this! 😉

**Reply by santicros** [reply]
I've tried to do my best here https://github.com/davidkpiano/xstate/pull/1991

Any feedback welcome :)
